sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -2666,3 +2666,54 @@ def test_dsl_pipeline_with_data_transfer_copy_2urifolder(self, client: MLClient)
         }
         assert expected_job == actual_job
 
+
+    @pytest.mark.disable_mock_code_hash
+    def test_register_output_sdk_succeed(self, client: MLClient):
+        component = load_component(source="./tests/test_configs/components/helloworld_component.yml")
+        component_input = Input(type='uri_file', path='https://dprepdata.blob.core.windows.net/demo/Titanic.csv')
+
+        @dsl.pipeline()
+        def sub_pipeline():
+            node = component(component_in_path=component_input)
+            node.outputs.component_out_path.name = 'sub_pipeline_output'
+            node.outputs.component_out_path.version = 'v1'
+            return {
+                'sub_pipeine_a_output': node.outputs.component_out_path
+            }
+
+        @dsl.pipeline()
+        def register_both_output():
```

Comment:
Add more tests to cover all output setting usage?

Scenario1

```Python
            node1.outputs.component_out_path = Output(
                name="name",
                version="version",
            )
```

Reply:
Scenario 2

```python
        @dsl.pipeline()
        def my_pipeline() -> Output(name="name", version="version"):
```

Reply:
will fix in next pr